### PR TITLE
Snmp trap receiver plugin

### PR DESCRIPTION
**Pull request by atolikin94**

### CODE REVIEW

File: plugins/snmp_receiver_plugin/snmp_server/snmp_parser.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+source = re.findall('UDP:.\[([\d.+]+)\]' ,r)
+
+# Building the HEADER so that Zabbix understands it
+HEADER = ''.join(source)
+IP = str(source).strip("[]'")
+HEADER = time + " MLNX SWITCH TRAP  " +  str(IP)
+
+# Appending it to the log file for Zabbix to pickup
+with open(destination, "a") as file:
+    file.write (HEADER + "\n" + r)
```
Comment (mike-dubman) [inline plugins/snmp_receiver_plugin/snmp_server/snmp_parser.py:28]: no file close?

File: plugins/snmp_receiver_plugin/snmp_server/start.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# (the "Company") and all right, title, and interest in and to the software
+# product, including all associated intellectual property rights, are and
+# shall remain exclusively with the Company.
+#
+# This software product is governed by the End User License Agreement
+# provided with the software product.
+#
+
+service snmptrapd start
+tail -f /dev/null
```
Comment (mike-dubman) [inline plugins/snmp_receiver_plugin/snmp_server/start.sh:15]: while needed? infinite loop?

File: plugins/snmp_receiver_plugin/snmp_server/pysnmp_receiver.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# to socket transport dispatcher
+snmpEngine = engine.SnmpEngine()
+
+# Transport setup
+
+# UDP over IPv4, first listening interface/port
+config.addTransport(
+    snmpEngine,
+    udp.domainName + (1,),
+    udp.UdpTransport().openServerMode(('0.0.0.0', 162))
```
Comment (mike-dubman) [inline plugins/snmp_receiver_plugin/snmp_server/pysnmp_receiver.py:15]: IMO IP:port should be configurable param

File: plugins/snmp_receiver_plugin/snmp_server/pysnmp_receiver.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+)
+
+# SNMPv1/2c setup
+
+# SecurityName <-> CommunityName mapping
+config.addV1System(snmpEngine, 'my-area', 'public')
```
Comment (mike-dubman) [inline plugins/snmp_receiver_plugin/snmp_server/pysnmp_receiver.py:21]: name should be configurable, with default ufm.clustername

File: plugins/snmp_receiver_plugin/build/config/deinit.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+# shall remain exclusively with the Company.
+#
+# This software product is governed by the End User License Agreement
+# provided with the software product.
+#
+
+set -eE
+LOG_FILE=/log/snmptrap.log
+if test -f "$LOG_FILE"; then
+    rm "$LOG_FILE"
```
Comment (alextabachnik) [inline plugins/snmp_receiver_plugin/build/config/deinit.sh:17]: could ask "are you sure you want to remove ?"
better to do rm -f or \rm

File: plugins/snmp_receiver_plugin/snmp_server/snmp_parser.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+time = time.strftime("%H:%M:%S %Y/%m/%d")
+time = str(time)
+
+# Matching on IPaddress
+source = re.findall('UDP:.\[([\d.+]+)\]' ,r)
+
+# Building the HEADER so that Zabbix understands it
+HEADER = ''.join(source)
+IP = str(source).strip("[]'")
+HEADER = time + " MLNX SWITCH TRAP  " +  str(IP)
```
Comment (alextabachnik) [inline plugins/snmp_receiver_plugin/snmp_server/snmp_parser.py:24]: in prev line you already did casting of IP to str

File: plugins/snmp_receiver_plugin/snmp_server/snmp_parser.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import re
+import sys
+import time
+
+# Trap write location:
+destination = "/log/snmptrap.log"  # File destination
```
Comment (alextabachnik) [inline plugins/snmp_receiver_plugin/snmp_server/snmp_parser.py:6]: May be the path should be configurable?

**Review by alextabachnik — APPROVED** [review]

**Comment by atolikin94** [comment]
Guys, thank you very much for your comments!
This is just a veeery raw POC, so probably I should have created a PR on such an early stage..
Will close it for now and re-open when it's reviewed carefully and tested!

**Pull request closed without merging**